mucore/src/muBase.c

```diff
@@ -1240,12 +1240,12 @@ MU_VOID muDebugError(muError_t errorcode)
 muError_t muDrawRectangle(muImage_t *SrcImg, muPoint_t p1, muPoint_t p2, MU_8S color)
 {
 	MU_32S i, j;
-	MU_32S offset, offset2;
+	MU_32S offset, offset2, step, step2;
 	MU_32S Ch_tmp;
 	MU_32S Ch_width;
 	MU_8U b = 0, g = 0, r = 0;
 	
-	if(SrcImg->channels != 3 || SrcImg->channels != 1)
+	if(SrcImg->channels != 3 && SrcImg->channels != 1)

```

Comment:
The OR is recommended due to this function just support 3 channels or 1 channel right?